src/test/java/pages/BasePage.java

```diff
@@ -0,0 +1,36 @@
+package pages;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+public class BasePage {
+    public WebDriver driver;
+    public WebDriverWait wait;
+
+    public BasePage(WebDriver driver){
+        this.driver = driver;
+        wait = new WebDriverWait(driver, 15);
+    }
+
+    public void waitVisibility(By element) {
+        wait.until(ExpectedConditions.visibilityOfAllElementsLocatedBy(element));
+    }
+
+    public void waitForClick(By element){
```

Comment:
Please, rename method (for example"waitElementVisibilityAndClick"), if you use this methods name , this method should only wait until element is clickable.